service/src/main/java/bio/terra/workspace/app/controller/ControlledAwsResourceApiController.java

```diff
@@ -206,6 +215,25 @@ public ResponseEntity<ApiAwsCredential> getAwsStorageFolderCredential(
       UUID resourceId,
       ApiAwsCredentialAccessScope accessScope,
       Integer duration) {
-    throw new NotImplementedException("not implemented");
+    final AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    controlledResourceMetadataManager
+        .validateControlledResourceAndAction(
+            userRequest, workspaceUuid, resourceId, getSamAction(accessScope))
+        .castByEnum(WsmResourceType.CONTROLLED_AWS_STORAGE_FOLDER);
+
+    Credentials awsCredentials =
```

Comment:
Discussed below, but this needs generate and pass the tags that allow access to the bucket as described in [the Principal Tags section of the design doc](https://docs.google.com/document/d/1qWFVAUwidqzx_mqda8vmkjjYnO3MvWwKi_w9z2n3mTA/edit#heading=h.ya9vyen240zq)...  these are echoed [here](https://github.com/verily-src/mc-terra-aws-wsm-landingzone/blob/3b03e739e4e01e5339c539909635fcbbbde59517/modules/environment/common.tf#L10)

(note the tag key names are a little stale, but let's solve one problem at a time and just go with them)